explorer/interpreter/interpreter.cpp

```diff
@@ -1456,7 +1456,16 @@ auto Interpreter::StepExp() -> ErrorOr<Success> {
         }
         case IntrinsicExpression::Intrinsic::Dealloc: {
           CARBON_CHECK(args.size() == 1);
-          heap_.Deallocate(cast<PointerValue>(args[0])->address());
+          const auto* ptr = cast<PointerValue>(args[0]);
+          if (act.pos() == 1) {
```

Comment:
```suggestion
          if (act.pos() == 0) {
```
It looks like this may deallocate _then_ read, i.e., use-after-free. I suspect this may not be caught by asan due to how data is managed. However, it indicates that either Deallocate isn't marking memory as invalid properly, or Read could maybe be validating in order to detect runtime errors. Maybe there's more that can be done here?

Reply:
`act.pos() == 0` is the first step for all instrinsics, where we process the parameters expression. So we will never enter this section with `0`. I will update the `else` to be clearer nonetheless.

Reply:
Maybe best to just add a CARBON_CHECK(act.pos() > 0) as enforcement? I'm not sure something like `else if (act.pos() == 2)` would be clear, not sure what you're intending.

Reply:
Yeah, that makes sense. Should be clearer and safer now.